sound/soc/sof/intel/cnl.c

```diff
@@ -280,6 +280,15 @@ int cnl_ipc4_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 	snd_sof_dsp_write(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDR,
 			  msg_data->primary | CNL_DSP_REG_HIPCIDR_BUSY);
 
+	/* Don't need to trigger d0i3_work for pm setting which is defined in module msg */
+	if (SOF_IPC4_MSG_IS_MODULE_MSG(msg_data->primary) &&
+	    (SOF_IPC4_MSG_TYPE_GET(msg_data->primary) == SOF_IPC4_MOD_SET_DX ||
+	    SOF_IPC4_MSG_TYPE_GET(msg_data->primary) == SOF_IPC4_MOD_SET_D0IX))
+		return 0;
+
+	mod_delayed_work(system_wq, &hdev->d0i3_work,
+			 msecs_to_jiffies(SOF_HDA_D0I3_WORK_DELAY_MS));
+
```

Comment:
what about `hda_dsp_ipc4_send_msg())` in `hda-ipc.c`?

Reply:
oh, APL was missed. thanks！